packages/theme/modules/checkout/composables/useCart/index.ts

```diff
@@ -197,21 +197,40 @@ PRODUCT
     }
   };
 
-  const applyCoupon = async ({ couponCode, customQuery }): Promise<void> => {
-    Logger.debug('useCart.applyCoupon');
+  const handleCoupon = async (couponCode = null, customQuery = null): Promise<boolean> => {
+    const variables = {
+      currentCart: cart.value,
+      customQuery,
+      couponCode,
+    };
 
-    try {
-      loading.value = true;
-      const { updatedCart } = await applyCouponCommand.execute(context, {
-        currentCart: cart.value,
-        couponCode,
-        customQuery,
-      });
+    const { updatedCart, errors } = couponCode
+      ? await applyCouponCommand.execute(context, variables)
+      : await removeCouponCommand.execute(context, variables);
 
-      error.value.applyCoupon = null;
+    if (errors) {
+      [error.value.applyCoupon] = errors;
```

Comment:
```suggestion
      error.value.applyCoupon = errors;
```
Not sure if my suggestion is correct. Can't wrap my head around whether this is detructuring?

Reply:
Because I want to get the first element. Apply coupon is not an array and I pretty much doubt that there will be more than one error in the same time..